Objects/listobject.c

```diff
@@ -80,9 +122,41 @@ list_resize(PyListObject *self, Py_ssize_t newsize)
 
     if (newsize == 0)
         new_allocated = 0;
+
+#ifdef Py_GIL_DISABLED
+    _PyListArray *array = NULL;
     if (new_allocated <= (size_t)PY_SSIZE_T_MAX / sizeof(PyObject *)) {
-        num_allocated_bytes = new_allocated * sizeof(PyObject *);
-        items = (PyObject **)PyMem_Realloc(self->ob_item, num_allocated_bytes);
+        array = list_allocate_array(new_allocated);
+    }
+    else {
+        // integer overflow
+        array = NULL;
+    }
```

Comment:
The size check here is redundant with the check in `list_allocate_array`. I think this can be:

```
_PyListArray *array = list_allocate_array(new_allocated);
```